### ISSUE TITLE
System.InvalidProgramException: Common Language Runtime detected an invalid program. (try-finally)

### ISSUE THREAD

**Issue by OwnageIsMagic** [ashmind/SharpLab#948]
On Run:
```
Exception
System.InvalidProgramException: Common Language Runtime detected an invalid program.
   at Prog.Main()
```

```c#
using System;
public class Prog
{
    public static void Main()
    {
        try
        {
            try
            {
                // caused by this
                throw new Exception("first");
            }
            finally
            {
                throw new Exception("second");
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine(ex.Message);
        }
    }
}
```

https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgZgAIMiAFMAewHM8BvPIpk9dR5h3Z7o4MAT3Y8inYcL6CuY7qOliA9PKIBjAIYQAzgFM4RAEb9eACwCWGoXO7AjVAO5EoW+wFEAHsq0AHYCcpQAFABEAGYmYBrAgQCUANwW0gC+8WKhUKoANumSlkyyOczWdg5ORG4e3r4BgdrKfnDRcVKWSU08LdJqwMpGRP5lXj5+RFquUcm54zyomACc/iMAdACyWhoaqtRasZPtzC0JQA=

**Comment by ashmind** [comment]
Thanks for reporting! This looks like an edge case in SharpLab rewriter -- I'll take a look (no specific timeline promise).

**Comment by OwnageIsMagic** [comment]
```c#
using System;
public class Prog
{
    public static void Main()
    {
        try
        {
            try
            {
                throw new Exception("first");
            }
            catch
            {
                throw new Exception("second");
            }
            // caused by this
            //*
            finally //*/
            {}
        }
        catch (Exception ex)
        {
            Console.WriteLine(ex.Message);
        }
    }
}
```

**Comment by ashmind** [comment]
Tested this while deploying fix for #1144, seems to be fixed by the same fix.

Note that your original link has a cached error and cannot be updated at the moment, please use this link to test:
https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgZgAIMiAFMAewHM8BvPIpk9dIx5h3Zno4MAJ4deRLiJH8h3cTzEzxAegVEAxgEMIAZwCmcIgCMBfABYBLTcPk9gxqgHciUbQ4CiADxXaADsFOUoABQARABmpmCawEEAlADcljIAvgniYVBqADYZUlZMcrnMNvaOzkTunj5+gUE6Kv5wMfHSVsnNvK0y6sAqxkQB5d6+/kTabtEpeRO8qJgAnAGjAHQAstqammrU2nFTHcytiUA
